internal_types/src/entry.rs

```
@@ -1329,6 +1346,174 @@ impl TryFrom<Vec<u8>> for SequencedEntry {
+        entries: &[Arc<OwnedSequencedEntry>],
+    ) -> Self {
+        let mut fbb = FlatBufferBuilder::new_with_capacity(1024);
+
+        let entries = entries
+            .iter()
+            .map(|e| {
+                let entry_bytes = fbb.create_vector_direct(
+                    e.fb()
+                        .entry_bytes()
+                        .expect("entry must be present in sequenced entry when initialized"),
+                );
+
+                entry_fb::SequencedEntry::create(
+                    &mut fbb,
+                    &entry_fb::SequencedEntryArgs {
+                        server_id: e.server_id().get_u32(),
+                        clock_value: e.clock_value().get_u64(),
+                        entry_bytes: Some(entry_bytes),
+                    },
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let entries = fbb.create_vector(&entries);
+
+        let segment = entry_fb::Segment::create(
+            &mut fbb,
+            &entry_fb::SegmentArgs {
+                id: segment_id,
+                server_id: server_id.get_u32(),
+                consistency_high_water_clock: clock_value.map(|c| c.get_u64()).unwrap_or(0),
+                entries: Some(entries),
+            },
+        );
+
+        fbb.finish(segment, None);
+
+        let (mut data, idx) = fbb.collapse();
+
+        Self::try_from(data.split_off(idx))
+            .expect("Flatbuffer data for sequenced entry just constructed should be valid")
+    }
+
+    /// Returns the Flatbuffers struct for the Segment
+    pub fn fb(&self) -> &entry_fb::Segment<'_> {
+        self.borrow_fb()
+    }
+
+    /// Returns the serialized bytes for the Segment
+    pub fn data(&self) -> &[u8] {
+        self.borrow_data()
+    }
+
+    pub fn consistency_high_water_clock(&self) -> Option<ClockValue> {
+        match self.fb().consistency_high_water_clock().try_into() {
+            Ok(c) => Some(c),
+            _ => None,
```

When would this be an error (or maybe it is an error when it is set?)

---

`consistency_high_water_clock()` returns `0` if the value is that or if it isn't present. The `try_into()` attempts to return the value into a ClockValue, which can't be 0. So just return None. I could have instead matched on the 0 value, but this seemed like the better thing to do to have the ClockValue type validate it?

---

It seems fine to me -- I was just wondering if there was some important-for-debugging error that was getting swallowed. Sounds like no